### ISSUE TITLE
Filename of WAFile created in (Zinc) server adaptor is not decoded

### ISSUE THREAD

**Issue by GoogleCodeExporter** [SeasideSt/Seaside#836]
```
What steps will reproduce the problem?
1. Use the Zinc Adaptor in Pharo (I tested with this one, but GemStone FastCGI 
also has this issue)
2. Upload a file with a filename using characters with codepoint > 256 with a 
modern browser and handle it in a Seaside file upload callback

What is the expected output? What do you see instead?
I expect the filename to be decoded. Now I see a string of the bytes in UTF8 
encoding.
The filename is part of the Content-Disposition header, which is UTF8 encoded 
by modern browsers [1][2].

[1] 
http://www.bennadel.com/blog/2591-embedding-foreign-characters-in-your-content-d
isposition-filename-header.htm
[2] http://greenbytes.de/tech/tc2231/

Proposed fix is the utf8 decoding of the filename when it is passed to the 
WAFile instance. I am currently assuming browsers always use UTF8 here, 
regardless of the encoding of the body (to be verified). Otherwise, 'self 
codec' should be used instead of the utf8 codec.

convertMultipartFileField: part  
	| file |
	(file := WAFile new)
		fileName: ((GRCodec forEncoding: 'utf-8') decode: part fileName);
		contentType: part contentType printString;
		contents: part contents.
	^ file
```

Original issue reported on code.google.com by `jo...@yesplan.be` on 7 Nov 2014 at 2:59

**Comment by johnnyzz** [comment]
The solution doesn't work for all characters,
```
(GRCodec forEncoding: 'utf-8') decode: 'hellò.doc'.
(GRCodec forEncoding: 'utf-8') decode: 'hellò.doc'.
(GRCodec forEncoding: 'utf-8') decode: 'í.txt'.
```
and it raises "Invalid utf8 input detected" error! (Firefox 41 under Linux)

**Comment by johnnyzz** [comment]
fileName decoding in `ZnZincServerAdaptor>>convertMultipartFileField:` should be removed, because it's already present in `ZnMimePart>>fileName` (with timestamp 10/10/2014), was added in `Zinc-Seaside-JohanBrichau.43` (14 November 2014)

**Comment by jbrichau** [comment]
Somehow, there has been a confusion because I had a different package for the same package version of Zinc-Seaside in my package cache. Moreover, it only seems to crash in Safari and Firefox and not in Chrome... See more in this thread: http://lists.squeakfoundation.org/pipermail/seaside/2016-June/032248.html

It should be fixed once Sven publishes the Zinc-Seaside-JohanBrichau.44 package and I updated the ConfigurationOfSeaside3 to reference it.

**Comment by jbrichau** [comment]
Fixed 3.2.1 and 3.1.6 loads.
